bids/grabbids/bids_validator.py

```diff
@@ -252,3 +256,129 @@ def conditional_match(self, expression, path):
+
+        config: string
+            Path to customized configuration file.
+
+
+    Examples
+    --------
+    >>> layout = bids.grabbids.BIDSLayout('/path/to/sample/project_root')
+    >>> validate_sequences(layout, '/path/to/sample/config.json')
+    >>> duplicate_file_df
+    # Put example output here
+    >>> summary_df
+    # Put example output here
+    >>> problem_df
+    # Put example output here
+    """
+    
+    duplicate_file_df = duplicate_check(layout)
+    summary_df, problem_df = expected_file_check(layout, config)
+    return duplicate_file_df, summary_df, problem_df
+    
+    
+def duplicate_check(layout):
+    """Checks images in BIDS project are not duplicated.
+
+    Check whether any files have duplicate content within the 
+    BIDS data set. Returns a data frame: duplicate_file_df.
+
+
+    Parameters
+    ----------
+        layout: BIDSLayout class
+            A BIDSLayout path of a data set.
+
+
+    Examples
+    --------
+    >>> layout = bids.grabbids.BIDSLayout('/path/to/sample/project_root')
+    >>> validate_sequences(layout, '/path/to/sample/config.json')
+    >>> duplicate_file_df
+    # Put example output here
+    """
+    
+    def md5(fname):
+        hash_md5 = hashlib.md5()
+        with open(fname, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+        return hash_md5.hexdigest()
+    hash_map = {}
+    all_niftis = layout.get(return_type="file", extensions='.nii.gz')
+    for nifti_file in all_niftis:
+        md5sum = md5(nifti_file)
+        if md5sum in hash_map:
+            hash_map[md5sum].append(nifti_file)
+        else:
+            hash_map[md5sum] = [nifti_file]
+    df = pd.DataFrame.from_dict(hash_map, orient='index') 
+    out_df = df.stack().reset_index().drop(columns='level_1').rename(columns={'level_0': 'hash', 0: 'filename'}) 
+    return out_df
+    
+    
+def expected_file_check(layout, config):
+    """Checks files in BIDS project match user defined expectations.
+
+    This method checks the number of data set files against a user customized 
+    configuration file. Returns two data frames: summary_df, problem_df.
+
+
+    Parameters
+    ----------
+        layout: BIDSLayout class
+            A BIDSLayout path of a data set.
+
+        config: string
+            Path to customized configuration file.
+
+
+    Examples
+    --------
+    >>> layout = bids.grabbids.BIDSLayout('/path/to/sample/project_root')
+    >>> validate_sequences(layout, '/path/to/sample/config.json')
+    >>> summary_df
+    # Put example output here
+    >>> problem_df
+    # Put example output here
+    """
+
+    summary_df = pd.DataFrame(columns=['subject', 'session', 'modality', 'task', 'runs', 'runs_found', 'problem'])
+    with open(config) as f:
+        json_data = json.load(f)
+        subjects = layout.get_subjects()
+    for sub in subjects: 
+        scan_params={}
+        for scan_params_d in json_data['entities']:
```

Comment:
Might be a good idea to rename the `'entities'` entry in the config to something else to avoid confusion (entities in pybids are individual keywords like `subject`, `run`, etc.). Maybe `'sequences'` or `'groups'` or something like that?